sqlglot/dialects/presto.py

```diff
@@ -224,8 +221,8 @@ class Generator(generator.Generator):
             exp.StructExtract: struct_extract_sql,
             exp.TableFormatProperty: lambda self, e: f"TABLE_FORMAT='{e.name.upper()}'",
             exp.FileFormatProperty: lambda self, e: f"FORMAT='{e.name.upper()}'",
-            exp.TimeStrToDate: _timestr_to_time_sql,
-            exp.TimeStrToTime: _timestr_to_time_sql,
+            exp.TimeStrToDate: timestrtotime_sql,
```

Comment:
This looks strange to me since we don't actually cast it into a date but this matches existing functionality.